src/components/editor/editor-preview.tsx

```diff
@@ -0,0 +1,179 @@
+  height: number;
+  gap: number;
+}
+
+/**
+ * Styled add question container component
+ */
+const AddQuestionContainer = styled(Stack, {
+  label: "add-question-container"
+})<AddQuestionContainerProps>(({ borderWidth, width, height, gap }) => ({
+  gap: gap,
+  textAlign: "center",
+  position: "absolute",
+  bottom: "10%",
+  right: "50%",
+  transform: "translateX(50%)",
+  backgroundColor: "#007f34",
+  borderWidth: borderWidth,
+  borderStyle: "dashed",
+  borderColor: "#fff",
+  display: "flex",
+  flexDirection: "column",
+  alignItems: "center",
+  justifyContent: "center",
+  width: width,
+  height: height
+}));
+
+/**
+ * Components properties
+ */
+interface Props {
+  htmlString: string;
+  width: number;
+  height: number;
+  scale: number;
+  page: Page;
+  selectedPage?: Page;
+  setSelectedPage: (pageId: string) => void;
+  onPanelPropertiesChange: () => void;
+  deletePage: (pageId?: string) => Promise<void>;
+  addQuestion: (questionType: PageQuestionType, pageId: string) => Promise<void>;
+}
+
+/**
+ * Editor preview component
+ */
+const EditorPreview = ({
+  htmlString,
+  width,
+  height,
+  scale,
+  page,
+  selectedPage,
+  setSelectedPage,
+  onPanelPropertiesChange,
+  deletePage,
+  addQuestion
+}: Props) => {
+  const [anchorEl, setAnchorEl] = useState<HTMLElement>();
+  const [pageMenuOpen, setPageMenuOpen] = useState(false);
+  const [addQuestionDialogOpen, setAddQuestionDialogOpen] = useState(false);
+
+  if (!page.id) return null;
+
+  /**
+   * Set up event listener to recieve post message from iframe
+   */
+  useEffect(() => {
+    // TODO: In the future when we need more event handlers within the iframe we can use a switch statement to replace the handlePostMessageEventListener.
+    // This can direct us to the appropriate event listener based upon the type of the Event e.g. IframeClickEvent, IframeButtonClickEvent etc.
+    window.addEventListener(`message-${page?.id}`, handlePostMessageEventListener);
+
+    return () => window.removeEventListener(`message-${page?.id}`, handlePostMessageEventListener);
+  }, []);
+
+  /**
+   * Handles post messsage event from iframe
+   *
+   * @param event message event
+   */
+  const handlePostMessageEventListener = (event: any) => {
+    const typedEvent = event as IframeClickEvent;
+
+    onPanelPropertiesChange();
+    setSelectedPage(typedEvent.detail.pageId);
+  };
+
+  /**
+   * Handles page menu click
+   *
+   * @param event event
+   */
+  const handlePageMenuClick = ({ target }: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(target as HTMLElement);
+    setPageMenuOpen(!pageMenuOpen);
+  };
+
+  return (
+    <div style={{ scale: `${scale}`, position: "relative" }}>
+      <MenuButton onClick={handlePageMenuClick}>
+        <MoreHoriz sx={{ width: "6rem", height: "6rem", color: "#ffffff" }} />
+      </MenuButton>
+      <iframe
+        srcDoc={wrapTemplate(parseHtmlToDom(htmlString).outerHTML, page.id)}
+        title="preview"
```

Comment:
Could this be localized?

Reply:
Actually why does it have a title called preview? It will appear when you hover the component. More useful in the editor would be to see ie. the type the page is or "Click page to edit".

Reply:
Dunno, the title was added like a month ago to mitigate linter warnings 😆

Reply:
Resolved